src/T4Generator/CustomCmdLineHost.cs

```diff
@@ -0,0 +1,304 @@
+			{
+				return false;
+			}
+		}
+		//Called by the Engine to enquire about   
+		//the processing options you require.   
+		//If you recognize that option, return an   
+		//appropriate value.   
+		//Otherwise, pass back NULL.  
+		//--------------------------------------------------------------------  
+		public object GetHostOption(string optionName)
+		{
+			object returnObject;
+			switch (optionName)
+			{
+				case "CacheAssemblies":
+					returnObject = true;
+					break;
+				default:
+					returnObject = null;
+					break;
+			}
+			return returnObject;
+		}
+		//The engine calls this method to resolve assembly references used in  
+		//the generated transformation class project and for the optional   
+		//assembly directive if the user has specified it in the text template.  
+		//This method can be called 0, 1, or more times.  
+		//---------------------------------------------------------------------  
+		public string ResolveAssemblyReference(string assemblyReference)
+		{
+			//If the argument is the fully qualified path of an existing file,  
+			//then we are done. (This does not do any work.)  
+			//----------------------------------------------------------------  
+			if (File.Exists(assemblyReference))
+			{
+				return assemblyReference;
+			}
+			//Maybe the assembly is in the same folder as the text template that   
+			//called the directive.  
+			//----------------------------------------------------------------  
+			string candidate = Path.Combine(Path.GetDirectoryName(this.TemplateFile), assemblyReference);
+			if (File.Exists(candidate))
+			{
+				return candidate;
+			}
+
+			//This can be customized to search specific paths for the file  
+			//or to search the GAC.  
+			//----------------------------------------------------------------  
+			//This can be customized to accept paths to search as command line  
+			//arguments.  
+			//----------------------------------------------------------------  
+			//If we cannot do better, return the original file name.  
+			return null;
+		}
+		//The engine calls this method based on the directives the user has   
+		//specified in the text template.  
+		//This method can be called 0, 1, or more times.  
+		//---------------------------------------------------------------------  
+		public Type ResolveDirectiveProcessor(string processorName)
+		{
+			//This host will not resolve any specific processors.  
+			//Check the processor name, and if it is the name of a processor the   
+			//host wants to support, return the type of the processor.  
+			//---------------------------------------------------------------------  
+			if (string.Compare(processorName, "XYZ", StringComparison.OrdinalIgnoreCase) == 0)
+			{
+				//return typeof();  
+			}
+			//This can be customized to search specific paths for the file  
+			//or to search the GAC  
+			//If the directive processor cannot be found, throw an error.  
+			throw new Exception("Directive Processor not found");
```

Comment:
Codacy found an issue: ['System.Exception' should not be thrown by user code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5863680)